adding a new suite using NOAH MP LSM for SR release.

panll opened the pull request:
Adding a new suite FV3_GSD_SAR_v1 for using NOAH LSM and turning off convection parameterization for high resolution.
The suite is tested in regional workflow.

climbfuji commented:
Is dtc/develop the correct target branch, or should that be master? Also, it would be good to add a regression test in both PROD and DEBUG mode. What do you think?

panll commented:
Either branch is fine to me. 
Yes, I can add the regression tests for it. 
Thanks!

climbfuji commented:
@llpcarson can you advise where this SDF and test should go in preparation for the SRW release? EMC develop+NCAR master or NCAR dtc/develop? Thanks!

llpcarson commented:
It will need to end up in EMC/develop for the SRW release branch.  I would think it makes sense to put it first into dtc/develop, to bundle with a commit at a later point, but you're (Dom) more aware of the work going on there.  And, yes, it needs a regression test (in addition to the regional_workflow).

climbfuji commented:
I am not sure if there is another slot for merging dtc/develop to master in the next weeks. Also, this PR is srw-specific. If we merge it into dtc/develop with the goal to merge it to master later, then we block a lot of development work that could otherwise happen in dtc/develop (because we can't commit it, or it would be sent to master at the same time).

Should this PR collect a bit more SRW work than just one SDF, and then merge it into master?

> On Apr 17, 2020, at 2:21 PM, Laurie Carson <notifications@github.com> wrote:
> 
> 
> It will need to end up in EMC/develop for the SRW release branch. I would think it makes sense to put it first into dtc/develop, to bundle with a commit at a later point, but you're (Dom) more aware of the work going on there. And, yes, it needs a regression test (in addition to the regional_workflow).
> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub <https://github.com/NCAR/fv3atm/pull/45#issuecomment-615446434>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB5C2RJSFED72575U44TEO3RNC23DANCNFSM4MK6IT6A>.
>

llpcarson commented:
I'm OK with this PR collecting more SRW updates and going directly to EMC/develop later, although hopefully there won't be other significant updates to the fv3atm repo needed for SRW - should know more in the next week or two.

panll commented:
added new suite RRFS_v0 with NOAH MP LSM for SRWeather release.
Modified file:
        gfsphysics/GFS_layer/GFS_typedefs.F90
added file:
       ccpp/suites/suite_FV3_RRFS_v0.xml

The suite is tested on hera.

climbfuji commented:
@panll do you want to update this PR as discussed last week?

panll commented:
Yes, thanks for asking! @climbfuji

panll commented:
suite_FV3_RRFS_v0.xml and GFS_typedefs.F90 are updated. @climbfuji

panll commented:
here is the suite file rrfs_v0/GSD_SAR_v1 in pull request. @JeffBeck-NOAA @gsketefian

JeffBeck-NOAA commented:
Thanks, Linlin!  Is this ready to merge?

climbfuji commented:
Waiting for the additional regression test using the SAR_v1 suite (ufs-weather-model PR), which requires hera to be back from maintenance for testing. Note these PRs go to NCAR dtc/develop. After they are in, I will update NCAR dtc/develop from EMC develop. We can collect SRW-App related changes in NCAR dtc/develop and send them back to EMC at some point later.

JeffBeck-NOAA commented:
Thanks, Dom.  FYI, in the near future, there are plans to move away from the NCAR fork and to use the ufs_community/ufs_weather_model develop branch directly.  I realize this may impact PR speed, since we'd be dealing directly with EMC in that case.

climbfuji commented:
> Thanks, Dom. FYI, in the near future, there are plans to move away from the NCAR fork and to use the ufs_community/ufs_weather_model develop branch directly. I realize this may impact PR speed, since we'd be dealing directly with EMC in that case.

I think this is a good idea. Using NCAR dtc/develop was good for the time being, especially when CCPP development happened only there. But now that CCPP is part of the main repository and developed by EMC and others, too, this move makes a lot of sense.

climbfuji commented:
Associated PRs:

https://github.com/NCAR/fv3atm/pull/45
https://github.com/NCAR/ufs-weather-model/pull/43
https://github.com/NCAR/ufs-weather-model/pull/49

For regression testing information, see https://github.com/NCAR/ufs-weather-model/pull/49.

JeffBeck-NOAA commented:
> Associated PRs:
> 
> #45
> [NCAR/ufs-weather-model#43](https://github.com/NCAR/ufs-weather-model/pull/43)
> [NCAR/ufs-weather-model#49](https://github.com/NCAR/ufs-weather-model/pull/49)
> 
> For regression testing information, see [NCAR/ufs-weather-model#49](https://github.com/NCAR/ufs-weather-model/pull/49).

Hi Dom, I think these commits are being moved to the ufs-community/ufs-weather-model develop branch.  @panll , right?

climbfuji commented:
No. They'll go to dtc/develop, followed by an update of dtc/develop from develop / master. dtc/develop will for the moment be the place to collect srw app related changes, so that we can update the authoritative branches in one commit. We can't keep EMC busy with sending small commits that require them to run the full suite of tests each time ... I am finalizing this as we speak.

> On Jun 5, 2020, at 1:27 PM, JeffBeck-NOAA <notifications@github.com> wrote:
> 
> 
> Associated PRs:
> 
> #45 <https://github.com/NCAR/fv3atm/pull/45>
> NCAR/ufs-weather-model#43 <https://github.com/NCAR/ufs-weather-model/pull/43>
> NCAR/ufs-weather-model#49 <https://github.com/NCAR/ufs-weather-model/pull/49>
> For regression testing information, see NCAR/ufs-weather-model#49 <https://github.com/NCAR/ufs-weather-model/pull/49>.
> 
> Hi Dom, I think these commits are being moved to the ufs-community/ufs-weather-model develop branch. @panll <https://github.com/panll> , right?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/NCAR/fv3atm/pull/45#issuecomment-639742945>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB5C2RLO63KQXWPPCJHNDM3RVFBLLANCNFSM4MK6IT6A>.
>

JeffBeck-NOAA commented:
OK, we had made a decision during our DTC UFS-CAM meeting this Wednesday to
move PRs and manage_externals to the authoritative repository.  It sounds
like that won't be possible and we'll need to reconsider.

On Fri, Jun 5, 2020 at 1:31 PM Dom Heinzeller <notifications@github.com>
wrote:

> No. They'll go to dtc/develop, followed by an update of dtc/develop from
> develop / master. dtc/develop will for the moment be the place to collect
> srw app related changes, so that we can update the authoritative branches
> in one commit. We can't keep EMC busy with sending small commits that
> require them to run the full suite of tests each time ... I am finalizing
> this as we speak.
>
> > On Jun 5, 2020, at 1:27 PM, JeffBeck-NOAA <notifications@github.com>
> wrote:
> >
> >
> > Associated PRs:
> >
> > #45 <https://github.com/NCAR/fv3atm/pull/45>
> > NCAR/ufs-weather-model#43 <
> https://github.com/NCAR/ufs-weather-model/pull/43>
> > NCAR/ufs-weather-model#49 <
> https://github.com/NCAR/ufs-weather-model/pull/49>
> > For regression testing information, see NCAR/ufs-weather-model#49 <
> https://github.com/NCAR/ufs-weather-model/pull/49>.
> >
> > Hi Dom, I think these commits are being moved to the
> ufs-community/ufs-weather-model develop branch. @panll <
> https://github.com/panll> , right?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub <
> https://github.com/NCAR/fv3atm/pull/45#issuecomment-639742945>, or
> unsubscribe <
> https://github.com/notifications/unsubscribe-auth/AB5C2RLO63KQXWPPCJHNDM3RVFBLLANCNFSM4MK6IT6A
> >.
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/fv3atm/pull/45#issuecomment-639745397>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANFE567HVNOG4O5WTHW6TSTRVFB2DANCNFSM4MK6IT6A>
> .
>

climbfuji commented:
> OK, we had made a decision during our DTC UFS-CAM meeting this Wednesday to move PRs and manage_externals to the authoritative repository. It sounds like that won't be possible and we'll need to reconsider. On Fri, Jun 5, 2020 at 1:31 PM Dom Heinzeller <notifications@github.com> wrote:
> […](#)

I am confused, but that's totally possible given my workload. Was I part of this meeting/decision? I remember we talked about moving over on the long run, but I don't recall immediately. Sorry for any misunderstanding. In any case we need to make more rapid updates while preparing for the SRW release, I suppose, and this will be difficult given the current commit queue for the authoritative repositories (we have one placeholder in there for committing = bringing a larger chunk of changes in, probably late June to July).

JeffBeck-NOAA commented:
Yes, this was an internal DTC UFS-CAM group decision made just this week after Julie had successfully moved the ufs-srweather-app umbrella repo to the authoritative ufs-weather-model develop branch.  We hadn't discussed any EMC burden regarding the PRs, but it sounds like it will be a significant hurdle.  Given your comments regarding the commit queue for the authoritative repo, we will definitely reconsider the timing of this change.  It sounds like it may be good to hold off for now.

climbfuji commented:
> Yes, this was an internal DTC UFS-CAM group decision made just this week after Julie had successfully moved the ufs-srweather-app umbrella repo to the authoritative ufs-weather-model develop branch. We hadn't discussed any EMC burden regarding the PRs, but it sounds like it will be a significant hurdle. Given your comments regarding the commit queue for the authoritative repo, we will definitely reconsider the timing of this change. We will make sure to consult with you in the future to determine when this might be optimal.

I am relieved that at least I didn't give you any contradictory information within three days! Let's keep in touch regarding the transition to determine the best timing as we head towards the SRW App release.

jwolff-ncar commented:
@climbfuji What is the cadence at which you update dtc/develop from the authoritative repo? We need to ensure we are update to date all around as we prepare for the release so we don't have a gotcha come up late in the game. Our preference is to work with the authoritative repo since we (at the moment) don't need a lot of changes to the weather model for this release. I could be oversimplifying this though and maybe there is a good reason not to move over yet? My understanding with where we are at now is that we are only missing the physics suite that @panll's PR addresses.  I am sure things will crop up, though.

climbfuji commented:
> @climbfuji What is the cadence at which you update dtc/develop from the authoritative repo? We need to ensure we are update to date all around as we prepare for the release so we don't have a gotcha come up late in the game. Our preference is to work with the authoritative repo since we (at the moment) don't need a lot of changes to the weather model for this release. I could be oversimplifying this though and maybe there is a good reason not to move over yet? My understanding with where we are at now is that we are only missing the physics suite that @panll's PR addresses. I am sure things will crop up, though.

We usually update with a few weeks time lag, unless there is a need for updating it quicker. The more often you update, the quicker it is. If need be, we can commit to a one-week delay at most until we are ready to transition the SRW folks to the authoritative repos (which should happen before the release, I definitely agree).

climbfuji commented:
> > @climbfuji What is the cadence at which you update dtc/develop from the authoritative repo? We need to ensure we are update to date all around as we prepare for the release so we don't have a gotcha come up late in the game. Our preference is to work with the authoritative repo since we (at the moment) don't need a lot of changes to the weather model for this release. I could be oversimplifying this though and maybe there is a good reason not to move over yet? My understanding with where we are at now is that we are only missing the physics suite that @panll's PR addresses. I am sure things will crop up, though.
> 
> We usually update with a few weeks time lag, unless there is a need for updating it quicker. The more often you update, the quicker it is. If need be, we can commit to a one-week delay at most until we are ready to transition the SRW folks to the authoritative repos (which should happen before the release, I definitely agree).

 Note that I'll make the next update right after these changes are merged (over the weekend).

climbfuji commented on ccpp/suites/suite_FV3_GSD_SAR_v1.xml:
```diff
@@ -0,0 +1,81 @@
+    <!-- End of surface iteration loop -->
+    <subcycle loop="1">
+      <scheme>GFS_surface_composites_post</scheme>
+      <scheme>dcyc2t3_post</scheme>
```
Please remove the line
```
<scheme>dcyc2t3_post</scheme>
```
this no longer exists.

climbfuji replied:
No need to fix this anymore, I did it when I merged your PRs.

climbfuji commented:
The NCAR dtc/develop code has been updated with the latest changes from the authoritative repositories as of today, and @panll's suites and regression tests were added.

JeffBeck-NOAA commented:
Thank you Dom for merging in the authoritative repo so quickly and Linlin
for getting these latest CCPP SDFs into the NCAR fork!

- Jeff

On Mon, Jun 8, 2020 at 8:14 AM Dom Heinzeller <notifications@github.com>
wrote:

> The NCAR dtc/develop code has been updated with the latest changes from
> the authoritative repositories as of today, and @panll
> <https://github.com/panll>'s suites and regression tests were added.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/fv3atm/pull/45#issuecomment-640641067>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANFE567LB3URD32U52STWHTRVTW2LANCNFSM4MK6IT6A>
> .
>
